vllm/v1/worker/gpu_input_batch.py

```diff
@@ -195,6 +195,10 @@ def __init__(
         self.logit_bias: List[Optional[Dict[int,
                                             float]]] = [None] * max_num_reqs
 
+        self.bad_words_token_ids: List[List[List[int]]] = [
+            list() for _ in range(max_num_reqs)
+        ]
```

Comment:
Could you change this to a `dict[int, List[List[int]]]`? Where the keys are req_index (not req_id), similar to `self.generators` above. This will minimize overhead for the 99.5% of cases where this isn't used.